samcli/lib/utils/stream_writer.py

```diff
@@ -1,28 +1,55 @@
 """
 This class acts like a wrapper around output streams to provide any flexibility with output we need
 """
-from typing import TextIO
+from io import BytesIO, TextIOWrapper
+from typing import Optional, TextIO, Union
 
 
 class StreamWriter:
-    def __init__(self, stream: TextIO, auto_flush: bool = False):
+    def __init__(self, stream: TextIO, stream_bytes: Optional[Union[TextIO, BytesIO]] = None, auto_flush: bool = False):
         """
         Instatiates new StreamWriter to the specified stream
 
         Parameters
         ----------
         stream io.RawIOBase
             Stream to wrap
+        stream_bytes io.TextIO | io.BytesIO
+            Stream to wrap if bytes are being written
         auto_flush bool
             Whether to autoflush the stream upon writing
         """
         self._stream = stream
+        self._stream_bytes = stream_bytes
         self._auto_flush = auto_flush
+        if isinstance(stream, TextIOWrapper):
+            self._stream_bytes = stream
```

Comment:
Nit:
```suggestion
        self._stream_bytes = stream if isinstance(stream, TextIOWrapper) else stream_bytes
        self._auto_flush = auto_flush
```
Maybe a one liner is a bit clearer here rather than assigning twice?